Solution v_1.0

bohdan-yatsyna opened the pull request.

bohdan-yatsyna committed:
Create and full in requirements.txt

bohdan-yatsyna committed:
Project initialize, create main.py

bohdan-yatsyna committed:
flake8 file create and fill in, additional modules install

bohdan-yatsyna committed:
Create db package with engine.py file

bohdan-yatsyna committed:
Refactor project structure according requirements

Delete db packaje, move files to root folder, engine.py rename to database.py

bohdan-yatsyna committed:
Create models.py with Author & Book models

bohdan-yatsyna committed:
Install & set up Alembic, make initial migration

bohdan-yatsyna committed:
Update .gitignore file with database

bohdan-yatsyna committed:
Fix problem with initial migration

bohdan-yatsyna committed:
Create Pydantic models in schemas.py

bohdan-yatsyna committed:
Little change in schemas.py

bohdan-yatsyna committed:
Create CRUD utility functions for authors & books in crud.py

bohdan-yatsyna committed:
Update main.py with all required endpoints & views

bohdan-yatsyna committed:
Fix N+1 problem for get_books()

bohdan-yatsyna committed:
Update .flake8 with ignore of VNE003

bohdan-yatsyna committed:
Fill in requirements.txt

ArsenPidhoretskyi commented on crud.py:
```diff
@@ -0,0 +1,69 @@
+    queryset = db.query(models.Book)
+
+    if author_id is not None:
+        return queryset.filter(models.Book.author_id == author_id).all()
```
```suggestion
        queryset = queryset.filter(models.Book.author_id == author_id)
```
Better to do like this. It will help to support code

ArsenPidhoretskyi commented on main.py:
```diff
@@ -0,0 +1,91 @@
+import schemas
+from database import SessionLocal
+
+app = FastAPI()
```
<img width="770" alt="image" src="https://github.com/mate-academy/py-fastapi-library-management-api/assets/65166203/0e500bf5-54ae-4b3c-a5c7-640850cbe7b9">

ArsenPidhoretskyi commented on crud.py:
```diff
@@ -0,0 +1,69 @@
+def get_authors(
+        db: Session,
+        skip: int = 0,
+        limit: int = 33,
```
Why limit is 33?)

ArsenPidhoretskyi replied:
As for me better to move default limit to separate constant and reuse it in all places

bohdan-yatsyna replied:
All your remarks noted with thanks and the code is fixed accordingly.

bohdan-yatsyna committed:
Little fixes in crud.py

Fix queryset in get_books_list_by_author_id, make default limit value in all functions with 100

bohdan-yatsyna committed:
Integrate SQLAlchemy models with main.py

bohdan-yatsyna committed:
Update migrations & requirements.txt

vsmutok reviewed (approved):
Good Job 👍
